### ISSUE TITLE
Typedef for structures makes stack view inconsistent

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#4822]
**Version and Platform (required):**
- Binary Ninja Version: 3.6.4712-dev Personal, 599e2ad7
- OS: macos
- OS Version: 14.1
- CPU Architecture: arm64

**Bug Description:**
In the image below you can see `UNICODE_STRING` and `struct _UNICODE_STRING` are actually same thing, but stack view does not thinks so.

**Steps To Reproduce:**
1. Create a structure and a typedef for it
2. Use that stucture and typedef in linear view
3. Open stack view see the bug

**Screenshots:**

<img width="367" src="https://github.com/Vector35/binaryninja-api/assets/109607896/4ad7802c-8424-4a52-9d65-d8734026cad9">

Typedef and structure:
```
typedef struct _UNICODE_STRING UNICODE_STRING;

struct _UNICODE_STRING
{
    USHORT Length;
    USHORT MaximumLength;
    PWCH Buffer;
};
```

**Comment by fuzyll** [comment]
If I'm understanding this correctly, you would prefer to see the entire structure in the stack view at both `-0x28` _and_ `-0x18`? I think that's a reasonable ask.

I do still think there should be a way to hide the internals of a struct in the stack view, but that doesn't need to be implemented as "make a `typedef`".

**Comment by op2786** [comment]
Yes, you understand it correctly. And I agree with there should be an option for hide the members of the structure.
